auth/providers/azure/pop_tokenverifier.go

```diff
@@ -193,6 +213,43 @@ func (p *PoPTokenVerifier) ValidatePopToken(token string) (string, error) {
 	return claims["at"].(string), nil
 }
 
+// verifyAccessTokenClaims verifies the inner access token has a cnf claim for
+// the PoP token public key.
+func (p *PoPTokenVerifier) verifyAccessTokenClaims(publicKey jwk, accessToken string) error {
+	// Parse JWT
+	accessTokenJwt, err := jwt.ParseSigned(accessToken)
+	if err != nil {
+		return fmt.Errorf("could not parse access token in PoP token: %w", err)
+	}
+
+	// Get claims
+	var claims Claims
+	err = accessTokenJwt.UnsafeClaimsWithoutVerification(&claims)
```

Comment:
can you add a comment on why it should be without verification?

Reply:
Updated